File: QLNH_Angular/QLNH_Angular/src/app/_models/item.ts
import { Category } from "./category";
import { Price } from "./price";
import { Restaurant } from "./restaurant";

export interface Item{
    id: number,
    name: string,
    description: string,
    created: Date,
    updated: Date,
    deleted: boolean,
    restaurantId: number,
    restaurant: Restaurant,
    categoryId: number,
    category: Category,
    prices: Price[]
}

File: QLNH_API/QLNH_Client/Models/Item.cs
using System.ComponentModel.DataAnnotations;

namespace QLNH_Client.Models
{
    public class Item
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double Discount { get; set; }
        public DateTime Created { get; set; }
        public DateTime Updated { get; set; }
        public bool Deleted { get; set; } = false;
        public virtual IList<Price>? Prices { get; set; }
        public virtual Restaurant? Restaurant { get; set; }
        public int? RestaurantId { get; set; }
        public virtual Category? Category { get; set; }
        public int? CategoryId { get; set; }
        public virtual IList<ItemImage>? ItemImages { get; set; }
    }
}
